source/registry/contracts/Registry.sol

```diff
@@ -90,13 +91,15 @@ contract Registry {
     uint256 length = supportedTokenList.length();
     require(length > 0, "NO_TOKENS_TO_REMOVE");
     address[] memory tokenList = new address[](length);
-    for (uint256 i = 0; i < length; i++) {
-      address token = supportedTokenList.at(0);
+        
+    for (uint256 i = length; i > 0;) {
+      i--;
```

Comment:
I tried doing the decrement in the condition like

```
for (uint256 i = length; i-- > 0;) {
  //
}
```

However the decrement didn't work inside the condition. Seems to be a bug with Solidity. So I do it inside the loop instead, which should be equivalent.